### PR TITLE
Revert "LibGit2: added tree related functions & tests"

### PR DESCRIPTION
**Pull request by tkelman**
Reverts JuliaLang/julia#16537
causing segfaults on osx travis, whoops
cc @wildart, anyone else have ideas what might be wrong?

### CODE REVIEW

File: base/libgit2/types.jl [hunk 1]
```diff
@@ -625,25 +625,24 @@ for (typ, ref, sup, fnc) in (
...
-        if $fnc !== nothing
-            function Base.finalize(obj::$typ)
-                if obj.ptr != C_NULL
-                    ccall(($fnc, :libgit2), Void, (Ptr{$ref},), obj.ptr)
-                    obj.ptr = C_NULL
-                end
+    end
+
+    if fnc !== nothing
+        @eval function Base.finalize(obj::$typ)
```
Comment (yuyichao) [inline base/libgit2/types.jl]: Unrelated to this PR but I don't think `Base.finalize` should be overloaded.

Comment (wildart) [inline base/libgit2/types.jl]: Would you care to elaborate?

Comment (yuyichao) [inline base/libgit2/types.jl]: `finalize` has a well defined meaning of calling the finalizer and remove that from the finalizer list. This overload doesn't do that.

**Comment by wildart** [comment]
I have no idea why macOS build fails on these tests.

**Comment by tkelman** [comment]
We can reopen a branch if that makes it easier for anyone to test and try to reproduce

**Comment by wildart** [comment]
I'll create branch in julia repo

**Pull request merged**